persistence/trees/trees.go

```diff
@@ -90,6 +91,8 @@ const (
 // of the underlying trees by utilizing the lazy loading
 // functionality provided by the underlying smt library.
 type treeStore struct {
+	modules.Module
+
 	treeStoreDir string
 	merkleTrees  map[merkleTree]*smt.SMT
 	nodeStores   map[merkleTree]kvstore.KVStore
```

Comment:
The fact that we still have `NewStateTrees` but also introduced `Create` seems like an anti-pattern: I don't know which one to use.

I think we should:
1.  rename `trees.go` to `module.go`
2. Consolidate all the business logic
3. Make the `dir` param part of the configs
4. Remove `NewStateTrees` altogether and use `Create` instead

Reply:
This is a good point about the clarity, I agree with refactoring the creation logic into the Create function. However, I would politely pushback on renaming trees.go to module.go, I think those two files should stay separate, trees.go should instead concern only the update and query functionality instead, and the creation logic it currently has should go in module.go.

Reply:
Pushback politely received and accepted 🤝